policy-packs/aws-advanced/compute.ts

```diff
@@ -160,30 +191,34 @@ export function requireDetailedMonitoring(name: string): Policy {
 // TODO: ec2-managedinstance-platform-check
 // https://docs.aws.amazon.com/config/latest/developerguide/ec2-managedinstance-platform-check.html
 
-export function requireEbsVolumesOnEc2Instances(name: string): Policy {
+export function requireEbsVolumesOnEc2Instances(name: string): ResourceValidationPolicy {
     // TODO: Check if EBS volumes are marked for deletion.
     return {
         name: name,
         description: "EBS volumes should be attached to all EC2 instances",
         enforcementLevel: "mandatory",
-        rules: typedRule(aws.ec2.Instance.isInstance, it =>
-            assert.ok(it.ebsBlockDevices === undefined || it.ebsBlockDevices.length > 0),
-        ),
+        validateResource: validateTypedResource(aws.ec2.Instance.isInstance, (it, args, reportViolation) => {
+            if (it.ebsBlockDevices !== undefined && it.ebsBlockDevices.length === 0) {
+                reportViolation("EC2 Instance should have EBS volumes attached.");
+            }
+        }),
     };
 }
 
 // TODO: eip-attached
 // https://docs.aws.amazon.com/config/latest/developerguide/eip-attached.html
 
-export function requireEbsEncryption(name: string, kmsKeyId?: string): Policy {
+export function requireEbsEncryption(name: string, kmsKeyId?: string): ResourceValidationPolicy {
     return {
         name: name,
         description: "EBS volumes should be encrypted",
         enforcementLevel: "mandatory",
-        rules: typedRule(aws.ebs.Volume.isInstance, it => {
-            assert.ok(it.encrypted);
-            if (kmsKeyId !== undefined) {
-                assert.strictEqual(it.kmsKeyId, kmsKeyId);
+        validateResource: validateTypedResource(aws.ebs.Volume.isInstance, (it, args, reportViolation) => {
+            if (!it.encrypted) {
+                reportViolation("EBS volumes should be encrypted.");
+            }
+            if (kmsKeyId !== undefined && it.kmsKeyId !== kmsKeyId) {
+                reportViolation(`EBS volumes should be encrypted with KMS ID '${kmsKeyId}'.`);
             }
```

Comment:
An example of where you might not want to necessarily return early (which happens with the assert case), but rather return as many violations as possible.